src/main/java/ee/carlrobert/codegpt/codecompletions/InlineCodeCompletionEditorTrackListener.java

```diff
@@ -0,0 +1,83 @@
+package ee.carlrobert.codegpt.codecompletions;
+
+import static java.util.stream.Collectors.toList;
+
+import com.intellij.codeInsight.daemon.impl.EditorTrackerListener;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.util.messages.MessageBusConnection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class InlineCodeCompletionEditorTrackListener implements EditorTrackerListener {
+
+  private static final Map<Project, InlineCodeCompletionEditorTrackListener> listeners =
```

Comment:
I changed the registrations of the `InlineCodeCompletionEditorTrackListener`, since I noticed multiple problems with the current enable/disable of the Code completion. The `activeEditorsChanged` is triggered even before any `StartupActivity` is triggered, so the 1st Code-Completion is requested before the TextEditor is really ready to use.
Also I encountered some problems when multiple projects were opened in Intellij.
You implemented the `Enable`-/`DisableCompletionsActions` which I quite like, but I thought it would be better if these actions really disable the entire `EditorInlayHandler` instead of keep triggering the `typingTimer` and simply returning immediately. IMO this approach now is much cleaner, if the code completion is disabled, no Timers or whatever are triggered.

Reply:
Hmm, what are the implications if `activeEditorsChanged` is called before `runActivity`?
`StartupActivity` needs to implement `DumbAware` so that it is triggered immediately after the project opens. However, I don't see any reason why we can't use completions beforehand.

Unfortunately, I'm not sure if this is a cleaner approach. Instead of letting the IDE to define the scope of the listener (application or project level), you're now handling essentially the same logic on your own. Perhaps it would be helpful if you could provide steps to reproduce the bug so that I can gain a better understanding of it.

Reply:
FYI: If you want to stop the timer upon status bar actions, you can create a new topic and subscribe to it in the `EditorInlayHandler`, where you can trigger the desired timer actions when a message is received.

Reply:
I added a new Topic for `codeCompletionEnabled` changes, the `EnableCompletionsAction`/`DisableCompletionsAction` publish messages accordingly, and the `InlineCodeCompletionEditorTrackListener`s subscribe to that topic. 
If code-completion is disabled, all `EditorInlayHandler`s are properly disposed and the `EditorTrackerEventListener` unsubscribed from the project's MessageBus.
If it is enabled, its subscribed to the project's MessageBus again for `CodeCompletionsEnabledListener`, and the `EditorInlayHandler` for the active Editor instantiated.
With this approach we dont have any unecessary Timers, ResetSuggestionListeners and `activeEditorsChanged` triggered when code-completion is disabled.

Reply:
Hmm, if the editor didn't load properly, or if the cursor wasn't visible, and the request was still triggered, then this likely indicates to a bug. I'm not entirely sure what will happen if we handle the completion logic on an indexed document, but in theory, all we need is the cursor's offset in the file. However, I agree with the rest, and we should trigger the logic only once the project is fully loaded.